test/Microsoft.AspNet.Mvc.Core.Test/DefaultAssemblyProviderTests.cs

```diff
@@ -164,5 +197,81 @@ public NullAssemblyProvider(ILibraryManager libraryManager) : base(libraryManage
             {
             }
         }
+
+        private class MvcAssembliesTestingProvider : DefaultAssemblyProvider
+        {
+            private static readonly ILibraryManager _libraryManager = GetLibraryManager();
+            private static readonly string _mvcName = "Microsoft.AspNet.Mvc";
+            private static IEnumerable<string> _assemblies; 
+
+            public MvcAssembliesTestingProvider() : base(_libraryManager)
+            { }
+
+            public HashSet<string> LoadedMvcReferenceAssemblies
```

Comment:
This seems really fragile. You want to get the library reference for `"Microsoft.AspNet.Mvc"` and then see what IT references. 

The goal is to **generate** the list that you have below in `ExpectedReferenceAssemblies`.